consoles/sshVirtsh.pm

```diff
@@ -489,48 +487,93 @@ sub define_and_start ($self) {
-    # just making sure we continue after the command finished
-    my $ignore = ' |& grep -v "\(failed to get domain\|Domain not found\)"';
-    $self->run_cmd("virsh $remote_vmm destroy " . $self->name . $ignore);
-    $self->run_cmd("virsh $remote_vmm undefine --snapshots-metadata " . $self->name . $ignore);
-
-    # define the new domain
-    $self->run_cmd("virsh $remote_vmm define $xmlfilename") && die "virsh define failed";
-    if ($self->vmm_family eq 'vmware') {
-        $self->run_cmd('echo bios.bootDelay = \"10000\" >> /vmfs/volumes/datastore1/openQA/' . $self->name . '.vmx', domain => 'sshVMwareServer');
-    }
+    $self->run_cmd("$self->{virsh} define $xmlfilename") && die "virsh define failed";
+}
 
-    $ret = $self->run_cmd("virsh $remote_vmm start " . $self->name . ' 2> >(tee /tmp/os-autoinst-' . $self->name . '-stderr.log >&2)');
+sub start_domain ($self) {
+    my $ret = $self->run_cmd("$self->{virsh} start $self->{name} 2> >(tee /tmp/os-autoinst-$self->{name}-stderr.log >&2)");
     bmwqemu::diag("Dump actually used libvirt configuration file " . ($ret ? "(broken)" : "(working)"));
-    $self->run_cmd("virsh $remote_vmm dumpxml " . $self->name);
+    $self->run_cmd("$self->{virsh} dumpxml $self->{name}");
     die "virsh start failed" if $ret;
 
     $self->backend->start_serial_grab($self->name);
+}
+
+sub shutdown_domain ($self) {
+    my $ignore = '|& grep -v "\(failed to get domain\|Domain not found\)"';
+    my $ret = $self->run_cmd("$self->{virsh} destroy $self->{name} $ignore");
+    bmwqemu::diag("virsh destroy returned $ret");
+}
+
+sub remove_domain ($self) {
+    my $ignore = '|& grep -v "\(failed to get domain\|Domain not found\)"';
+    my $ret = $self->run_cmd("$self->{virsh} undefine $self->{name} --snapshots-metadata $ignore");
+    bmwqemu::diag("virsh undefine returned $ret");
+}
+
+sub modify_vmx ($self) {
+    # VMX files are VMWare specific
+    return unless ($self->vmm_family eq 'vmware');
+
+    my $vmx = sprintf('/vmfs/volumes/datastore1/openQA/%s.vmx', $self->name);
+    my $ci_meta = $bmwqemu::vars{CLOUD_INIT_META};
+    my $ci_user = $bmwqemu::vars{CLOUD_INIT_USER};
+
+    # set default boot delay
+    $self->run_cmd("echo bios.bootDelay = \"10000\" >> $vmx", domain => 'sshVMwareServer');
+
+    # inject cloud init metadata and userdata required for the image
+    if ($ci_meta && $ci_user) {
+        $self->run_cmd("echo guestinfo.metadata = \"$ci_meta\" >> $vmx", domain => 'sshVMwareServer');
+        $self->run_cmd("echo guestinfo.metadata.encoding = \"gzip+base64\" >> $vmx", domain => 'sshVMwareServer');
+        $self->run_cmd("echo guestinfo.userdata = \"$ci_user\" >> $vmx", domain => 'sshVMwareServer');
+        $self->run_cmd("echo guestinfo.userdata.encoding = \"gzip+base64\" >> $vmx", domain => 'sshVMwareServer');
+    }
+}
+
+sub define_and_start ($self) {
+    bmwqemu::diag("Method consoles::sshVirtsh::define_and_start will become DEPRECATED!");
```

Comment:
Why deprecating it? This function is used for other VM types (besides VMWare). I'd generally try to minimize the impact of your change for other VM types. (If you really need to affect other VM types, you should also test these, see https://github.com/os-autoinst/os-autoinst/blob/master/doc/backends.md for some documentation.)

Reply:
Keep in mind that we can only merge your changes when `xen` and `zkvm` work still unaffected. So generally deprecating this function seems still wrong. If you need to do specific things for `vmware` then just do that but keep other code in place.

Reply:
I am not going to remove the function within this PR, just marking that it might be removed in the future. However, if this is not recommended, let's keep it then.

Reply:
Discussed it briefly. As I understand it there is no plans to refactor other backends soon, so I suggest to drop the deprecation message (which would be misleading) and just keep the changes minimal. Also, double-check that coverage looks okay e.g. at least hits the code path in existing tests.